pdns/auth-domaincache.cc

```diff
@@ -0,0 +1,166 @@
+/*
+ * This file is part of PowerDNS or dnsdist.
+ * Copyright -- PowerDNS.COM B.V. and its contributors
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of version 2 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ *
+ * In addition, for the avoidance of any doubt, permission is granted to
+ * link this program with OpenSSL and to (re)distribute the binaries
+ * produced as the result of such linking.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "auth-domaincache.hh"
+#include "logger.hh"
+#include "statbag.hh"
+#include "arguments.hh"
+#include "cachecleaner.hh"
+extern StatBag S;
+
+AuthDomainCache::AuthDomainCache(size_t mapsCount) :
+  d_maps(mapsCount)
+{
+  S.declare("domain-cache-hit", "Number of hits on the domain cache");
+  S.declare("domain-cache-miss", "Number of misses on the domain cache");
```

Comment:
```suggestion
  S.declare("domain-cache-hit", "Number of hits of the domain cache");
  S.declare("domain-cache-miss", "Number of misses of the domain cache");
```
?

Reply:
`on` is better than `of`. `in` might be better than `on`.

Also, `on` is consistent with existing entries.

Reply:
```suggestion
  S.declare("domain-cache-hit", "Number of hits on the domain cache");
  S.declare("domain-cache-miss", "Number of domain cache misses");
```

Reply:
(and similar for the other line if this seems like a good style)

Reply:
This makes sense to me, but then we should also change the existing similar entries.

Reply:
I've done it here, but the other places need to be in a separate PR

Reply:
agreed